### PR TITLE
Release

### PR DESCRIPTION
**Pull request by fbwoolf**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/stacks-wallet-web/actions/runs/1515118802).

PR for release.

cc/ @kyranjamie @markmhx

### CODE REVIEW

**Review by markmhendrickson — APPROVED** [review]

**Comment by fbwoolf** [comment]
@markmhx not totally sure where to go from here. Sorry, a bit confused by this process atm.

**Comment by markmhendrickson** [comment]
All good, I've submitted this build manually to Chrome as `2.21.4`. We can release to Firefox whenever @kyranjamie has the chance to help with this merge.

**Review by kyranjamie — APPROVED** [review]

**Comment by kyranjamie** [comment]
Not entirely sure what's going to happen if `v2.21.4` as published manually. CI will make and try to release `v2.21.3`. Have a pretty busy agenda today so reluctant to merge this rn, but with the warning in place I don't think this isn't super urgent?.

**Comment by markmhendrickson** [comment]
Not urgent, let's wait until Monday. Do we think it might have an issue with skipping sub-versions?

File: src/common/transactions/transaction-utils.ts [hunk 1]
```diff
@@ -49,6 +49,32 @@ function txHasTime(tx: Tx) {
...
+  if (new BigNumber(feeEstimations[1].fee).isGreaterThan(maxValueForMiddleFeeEstimation)) {
+    feeEstimationsWithMaxValues.push({ fee: maxValueForMiddleFeeEstimation, fee_rate: 0 });
+  } else {
+    feeEstimationsWithMaxValues.push(feeEstimations[1]);
+  }
+  if (new BigNumber(feeEstimations[2].fee).isGreaterThan(maxValueForHighFeeEstimation)) {
+    feeEstimationsWithMaxValues.push({ fee: maxValueForHighFeeEstimation, fee_rate: 0 });
+  } else {
+    feeEstimationsWithMaxValues.push(feeEstimations[2]);
+  }
```
Comment (kyranjamie) [inline src/common/transactions/transaction-utils.ts:73]: Alternate implementation might look something like this:

```suggestion
const maxHighFeeEstimations = [
  maxValueForLowFeeEstimation,
  maxValueForMiddleFeeEstimation,
  maxValueForHighFeeEstimation,
];

export function getFeeEstimationsWithMaxValues(feeEstimations: FeeEstimation[]) {
  return feeEstimations.map(({ fee, ...rest }, index) => ({
    ...rest,
    fee: BigNumber.min(fee, maxHighFeeEstimations[index]),
  }));
}
```

Mapping an array to get the result, using `BigNumber.min` to select the lower of the two values.

**Comment by fbwoolf** [comment]
> 

The last release was done the same way. @markmhx had manually submitted 2.21.3 to Chrome and then I merged a release PR after and it automatically submitted 2.21.2 to Firefox.

File: src/common/transactions/transaction-utils.ts (hunk 1 above)
Comment (fbwoolf) [reply src/common/transactions/transaction-utils.ts:73]: Ok, thanks for the suggestion. Will keep in mind for next time.

**Comment by markmhendrickson** [comment]
I've just submitted version `2.21.5` to Chrome manually with downward cap adjustments to 0.5, 0.75, and 2 STX for low, standard and high estimates respectively. [Per Discord notice](https://discord.com/channels/621759717756370964/745197302255321108/914225754294870086).

Let's be sure to update this PR with those new values before merge and automatic release to Chrome and Firefox.

**Comment by fbwoolf** [comment]
> I've just submitted version `2.21.5` to Chrome manually with downward cap adjustments to 0.5, 0.75, and 2 STX for low, standard and high estimates respectively. [Per Discord notice](https://discord.com/channels/621759717756370964/745197302255321108/914225754294870086).
> 
> Let's be sure to update this PR with those new values before merge and automatic release to Chrome and Firefox.

You should definitely ping me if you need help updating things. I didn't see this posted until now.

**Comment by markmhendrickson** [comment]
@beguene @kyranjamie can one of you this morning update the fee values in this PR per https://github.com/hirosystems/stacks-wallet-web/pull/2001#issuecomment-980786778 and merge into main for release to Firefox in particular?

**Comment by beguene** [comment]
@markmhx This has to be done on a different branch first. Then we can merge it back.

**Comment by markmhendrickson** [comment]
@beguene That works for me, can you create the separate branch etc.?

**Comment by beguene** [comment]
> @beguene That works for me, can you create the separate branch etc.?

Done here https://github.com/hirosystems/stacks-wallet-web/pull/2003
Need a check by @kyranjamie then we can merge.

**Pull request merged**